File: fizzbuzz.ml
let doit i = (match (i mod 3, i mod 5) with
    | (0, 0) -> print_string "FizzBuzz"
    | (0, _) -> print_string "Fizz"
    | (_, 0) -> print_string "Buzz"
    | _ -> print_int i); print_newline ()
;;

for i = 1 to 100 do
        doit i
done

File: fizzbuzz.lua
for i=1,100 do
    local r3, r5 = i % 3, i % 5
    if r3 == 0 then io.stdout:write "Fizz" end
    if r5 == 0 then io.stdout:write "Buzz" end
    if (r3 * r5) ~= 0 then io.stdout:write(i) end
    print()
end
